foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiClient.java

```diff
@@ -105,6 +108,41 @@ public OpenAiChatCompletionOutput chatCompletion(
     return execute("/chat/completions", parameters, OpenAiChatCompletionOutput.class);
   }
 
+  /**
+   * Generate a completion for the given prompt.
+   *
+   * @param parameters the prompt, including messages and other parameters.
+   * @return A stream of chat completions deltas
+   * @throws OpenAiClientException if the request fails
+   */
+  @Nonnull
+  public Stream<OpenAiChatCompletionDelta> streamChatCompletion(
+      @Nonnull final OpenAiChatCompletionParameters parameters) throws OpenAiClientException {
+    parameters.setStream(true);
+    return streamChatCompletion("/chat/completions", parameters, OpenAiChatCompletionDelta.class);
+  }
+
+  /**
+   * Generate a completion for the given prompt.
+   *
+   * @param parameters the prompt, including messages and other parameters.
+   * @return A stream of message deltas
+   * @throws OpenAiClientException if the request fails or if the finish reason is content_filter
+   */
+  @Nonnull
+  public Stream<String> simpleStreamChatCompletion(
```

Comment:
How about calling this:

```suggestion
  public Stream<String> streamChatCompletion(
```

and the other one `streamChatCompletionDeltas`?